microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/services/MicroProfileValidator.java

```diff
@@ -262,6 +266,24 @@ private static boolean isShortString(String str) {
 		}
 	}
 
+	private static boolean isBigDecimalString(String str) {
+		try {
+			new BigDecimal(str);
+			return true;
+		} catch (NumberFormatException e) {
+			return false;
+		}
+	}
+
+	private static boolean isBigIntegerString(String str) {
+		try {

```

Comment:
check if str is null or empty before. No need to throw an exception if it can be avoided (exception creation is expensive)

Reply:
fixed